### PR TITLE
Fix status title for statuses without text

### PR DESCRIPTION
**Pull request by ghost**
I fixed a bug that is visible when a post has no text and no content warnings. In that case, it displays `<user>: (no text was provided) - <server-title>` instead of `<user>: "" - <server-title>`. I would appreciate feedback as this is my first direct contribution to Mastodon, please. Thank you,

### CODE REVIEW

**Comment by trwnh** [comment]
it might be a good idea to instead do something like `(1 attachment)`?

**Comment by IRod22** [comment]
> it might be a good idea to instead do something like `(1 attachment)`?

@trwnh, that's a good idea, but I'm going to need to do some digging to see 1) what the `status` map contains, and 2) the attachment structure.

**Comment by trwnh** [comment]
i think it should follow the Status entity structure? in which case it'd be `media_attachments[].length` https://docs.joinmastodon.org/entities/Status/

**Comment by IRod22** [comment]
@trwnh, although I appreciate the suggestion, the `status` map is more complex than I anticipated, and I don't want to break anything while trying to see how it works, How about `(contains attachments)` instead?

**Comment by trwnh** [comment]
i suppose that could work too -- `(media attached)`

**Comment by IRod22** [comment]
@trwnh I have the change ready to go; now, we just sit here and wait until someone can review this.

**Comment by IRod22** [comment]
🤔 I hope this will get reviewed and merged soon. I don't feel confident about my rebasing skills, and I don't want a merge conflict on my hands.

File: app/javascript/mastodon/features/status/index.jsx
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/status/index.jsx:176]: Hi! All UI text should be translatable (see `intl.formatMessage` calls). In addition, you can use `status.get('media_attachments').size` to know how many attachments the post has.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

File: app/javascript/mastodon/features/status/index.jsx
Comment (IRod22) [reply app/javascript/mastodon/features/status/index.jsx]: @ClearlyClaire, I added the required `en` and `es-MX` translations, and I translated the status title.

File: app/javascript/mastodon/features/status/index.jsx
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/status/index.jsx:75]: I'm not convinced by the new format, I think “user: truncated post text - site name” was fine…

For consistency, the format for text-less posts could be “user posted N media attachments - site name” for consistency.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

File: app/javascript/mastodon/features/status/index.jsx
Comment (IRod22) [reply app/javascript/mastodon/features/status/index.jsx:75]: @ClearlyClaire, I implemented the title like this, so it's more human readable and adaptable to the server's or user's locale. To see what I mean, look at the tab title for this PR and notice how it has a similar structure to the status title I implemented. Also, I decided to implement the ellipsis as a separate translation because it can be written differently in different languages.

**Comment by IRod22** [comment]
@ClearlyClaire I fixed the code regarding `intl`, but I would like to see my comment regarding the title format, please.

**Comment by IRod22** [comment]
@ClearlyClaire I fixed a mistake I made in a previous commit.

**Comment by IRod22** [comment]
Fat fingered. 🤦‍♂️

**Comment by IRod22** [comment]
@ClearlyClaire, I fixed an error from ESLint, and the code is ready for review.

**Review by ClearlyClaire — APPROVED** [review]
Apart from the new string missing from `defaultMessages.json`, this looks good to me! Thanks for your contribution!

I understand your reasons for initially changing the title format, but I think the current format does not need changing. That can be discussed and possibly changed in a later PR, though.

**Comment by IRod22** [comment]
@ClearlyClaire, @nschonni, [this is taking longer than I expected](https://urusai.social/@ivan18rod/110195196745169369).

**Comment by ClearlyClaire** [comment]
@IRod22 I am unfamiliar with developing on Windows, but I can run `manage:translations` on my end and push to your branch if that'd help.

**Comment by IRod22** [comment]
@ClearlyClaire, that would really help. Thanl you. 🙌

**Comment by ClearlyClaire** [comment]
Well, seems like I don't have write access to the branch, I thought GitHub allowed that for PRs, but I may be misremembering…

In any case, the patch you want to apply is:
```patch
diff --git a/app/javascript/mastodon/locales/defaultMessages.json b/app/javascript/mastodon/locales/defaultMessages.json
index d8f40aa510..6bb6ce42b0 100644
--- a/app/javascript/mastodon/locales/defaultMessages.json
+++ b/app/javascript/mastodon/locales/defaultMessages.json
@@ -3724,6 +3724,10 @@
         "defaultMessage": "Show less for all",
         "id": "status.show_less_all"
       },
+      {
+        "defaultMessage": "{user} posted {attachmentCount, plural, one {an attachment} other {{attachmentCount} attachments}}",
+        "id": "status.title.with_attachments"
+      },
       {
         "defaultMessage": "Detailed conversation view",
         "id": "status.detailed_status"
```

**Comment by ClearlyClaire** [comment]
This is a ready-to-apply patch, you should not look for the line manually, but use the `patch` command-line tool instead!

**Comment by IRod22** [comment]
@ClearlyClaire Got it! My bad.

**Comment by IRod22** [comment]
@ClearlyClaire, the patch failed:

```
> Get-Clipboard | git apply
error: patch failed: app/javascript/mastodon/locales/defaultMessages.json:3724
error: app/javascript/mastodon/locales/defaultMessages.json: patch does not apply
>
```

I tried finding the `patch` command, but I could not find it (it's probably on unix only), so I had to use `git apply` and get the copied patch using `Get-Clipboard`.

**Comment by ClearlyClaire** [comment]
The following should be formatted properly for git:

```patch
From ad2d2625884d803b008facc7ece6b6723c3eea8c Mon Sep 17 00:00:00 2001
From: Claire <claire.github-309c@sitedethib.com>
Date: Fri, 14 Apr 2023 16:41:38 +0200
Subject: [PATCH] Add new translatable string to `defaultMessages.json`

---
 app/javascript/mastodon/locales/defaultMessages.json | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/app/javascript/mastodon/locales/defaultMessages.json b/app/javascript/mastodon/locales/defaultMessages.json
index d8f40aa510..6bb6ce42b0 100644
--- a/app/javascript/mastodon/locales/defaultMessages.json
+++ b/app/javascript/mastodon/locales/defaultMessages.json
@@ -3724,6 +3724,10 @@
         "defaultMessage": "Show less for all",
         "id": "status.show_less_all"
       },
+      {
+        "defaultMessage": "{user} posted {attachmentCount, plural, one {an attachment} other {{attachmentCount} attachments}}",
+        "id": "status.title.with_attachments"
+      },
       {
         "defaultMessage": "Detailed conversation view",
         "id": "status.detailed_status"
-- 
2.39.2

```

**Comment by IRod22** [comment]
@ClearlyClaire, the patch worked, but a pre-commit hook is preventing a commit because of a syntax error:

```
Syntax Error: Expected ":" or blank space but "\n" found.
```

**Comment by ClearlyClaire** [comment]
What file is that in? Do you have any uncommitted change?

**Comment by IRod22** [comment]
It looks like the i18n workflow works!

**Review by ClearlyClaire — APPROVED** [review]

**Comment by IRod22** [comment]
Thank you, @ClearlyClaire! Do you mind if I ask what your mastodon handle is?

**Pull request merged**
